galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.cloud.manager/src/main/java/dev/galasa/cloud/ICloudContainer.java

```diff
@@ -0,0 +1,65 @@
+/*
+* Copyright contributors to the Galasa project 
+*/
+package dev.galasa.cloud;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Cloud Container
+ * 
+ * @author Michael Baylis
+ *
+ */
+public interface ICloudContainer {
+	
+	/**
+	 * Retrieve the port number for an exposed port
+	 * 
+	 * @param portName - The name of the port
+	 * @return the hostname/port
+	 * @throws CloudManagerException
+	 */
+	public InetSocketAddress getContainerExposedPort(String portName) throws CloudManagerException;
+	/**
+	 * Retrieve the port number for an exposed port
+	 * 
+	 * @param portNumber - The number of the port
+	 * @return the hostname/port
+	 * @throws CloudManagerException
+	 */
+	public InetSocketAddress getContainerExposedPort(int portNumber) throws CloudManagerException;
+
+	/**
+	 * Retrieve the stdout/stderr log of the container
+	 * 
+	 * @return The log
+	 * @throws CloudManagerException
+	 */
+	public String getLog() throws CloudManagerException;
+	
+	/**
+	 * Retrieve the contents of a file within the container
+	 * 
+	 * @param path - The absolute path of the file
+	 * @return The contents of the file
+	 * @throws CloudManagerException
+	 */
+	public String retrieveFileAsString(String path) throws CloudManagerException;
+
```

Comment:
Could it be possible to have an exec for the container? This could be useful in applications that have CLI's build into the container. E.g etcd container might want to do a `etcdctl get --prefix=true "" > somefile.txt` and retrieve the output above, or even the direct out from the command.

Reply:
Yeah, my thoughts were at an application level rather than an individual container feature. Maybe my example was bad and its impossible to know the shape of peoples applciations, but I was thinking that people may have checks they run to ensure that one service can talk to another in many different ways. The context in my head is about people who may be looking to automate steps that they do manually. AKA i bring up X server and Y server then make sure they can communicate (which may not just be log driven)

Reply:
if we can add it easily later, we will,   but not until after education and getting the big 3 apis complete.

Reply:
Ok. Yesh most things are done via logs and this is something that could be impl-ed if requested from someone. I do like the idea of people using this manager as part of thier application manager and being able to do any setup/validation steps that thier application requires is all

Reply:
ok,  you have made your point.   BUT until I have fully researched the big 3 and got it working,   I am not going to add this.   Our priority here is Education,  not providing a full blown Cloud Manager,  as that could take months.
So for the moment,  you point is noted,  and overridden.